pkg/build/builder/common.go

```diff
@@ -201,13 +203,32 @@ func retryBuildStatusUpdate(build *buildapi.Build, client client.BuildInterface,
 		latestBuild.Status.Output.To = build.Status.Output.To
 		latestBuild.Status.Stages = build.Status.Stages
 
-		if _, err := client.UpdateDetails(latestBuild); err != nil {
+		if err = retryBuildDetailsUpdate(latestBuild, client, 5); err != nil {
 			return err
 		}
+
 		return nil
 	})
 }
 
+// retryBuildDetailsUpdate retries the build status update in case of a non-conflict error
+// retrying all non-conflict errors here as we don't expect to encounter a fatal error
+func retryBuildDetailsUpdate(latestBuild *buildapi.Build, client client.BuildInterface, times int) error {
+	_, err := client.UpdateDetails(latestBuild)
+	if err != nil {
+		if errors.IsConflict(err) || times <= 0 {
+			return err
+		} else {
+			glog.V(4).Infof("retrying build details update: %v", err)
+			time.Sleep(1 * time.Second)
+			return retryBuildDetailsUpdate(latestBuild, client, times-1)
```

Comment:
i don't want to blow up the stack when someone passes retry times==MAX_INT64, please make it a loop